src/components/Dataset/DatasetPageHeader.js

```diff
@@ -32,8 +32,7 @@ const Block = ({ children }) => {
   )
 }
 
-export const DatasetPageHeader = ({ dataset }) => {
-  const { error } = useDatasetManager()
+export const DatasetPageHeader = ({ dataset, hasError }) => {
```

Comment:
I'm trying to understand why we must pass in `hasError` and wouldn't be able to derive it from the `dataset` prop.

Reply:
The `useResourceLoader` hook is used by different components (including this `DatasetPageHeader`, [DownloadNowModal](https://github.com/AlexsLemonade/refinebio-web/blob/b45ba3d342e247f094c3ece4bcb79bc509e5a437/src/components/shared/SearchCard/SearchCardAction/DownloadNowModal.js#L24-L26), and the [experiment](https://github.com/AlexsLemonade/refinebio-web/blob/b45ba3d342e247f094c3ece4bcb79bc509e5a437/src/components/shared/SearchCard/SearchCardAction/DownloadNowModal.js#L24-L27) page components) to "watch" the latest processing status of resources (a user-custom dataset and an one-of experiment download's dataset). The `hasError` state is set to true if API requests fail while polling the data, and it could be used to display a error message in UI etc.  

In terms of the `dataset` state, it's a part of `useDatasetManager` and is used to render the dataset's data in UI.

This implementation (utilizing the `hasError` state) also matches other hooks' pattern ([`useComendia`](https://github.com/AlexsLemonade/refinebio-web/blob/118c281d226a70d6cdc972c290b87b25c93c6b18/src/hooks/useCompendia.js#L20), [`useLoader`](https://github.com/AlexsLemonade/refinebio-web/blob/118c281d226a70d6cdc972c290b87b25c93c6b18/src/hooks/useLoader.js#L10), and [`useSamplesTable`](https://github.com/AlexsLemonade/refinebio-web/blob/118c281d226a70d6cdc972c290b87b25c93c6b18/src/hooks/useSamplesTableManager.js#L17), and the helper [`fetchSearch`](https://github.com/AlexsLemonade/refinebio-web/blob/118c281d226a70d6cdc972c290b87b25c93c6b18/src/helpers/fetchSearch.js#L62) for search results page that fetches API data via its `getInitialProps`). 

For this, I used the `useResourceLoader` hook's `hasError` (instead of `useDatasetManager`'s `error` object - please see the PR https://github.com/AlexsLemonade/refinebio-web/pull/297) and pass it to `DatasetPageHeader` (which is responsible for re-rendering UI based on the most recent dataset status that was last polled from API) via the [`dataset_id`](https://github.com/AlexsLemonade/refinebio-web/blob/b45ba3d342e247f094c3ece4bcb79bc509e5a437/src/pages/dataset/%5Bdataset_id%5D.js#L33-L38) page component.